### DISCUSSION TITLE
Message passing in NNConv

### DISCUSSION THREAD

**Discussion by hkim716** [pyg-team/pytorch_geometric#4858] [Q&A] [answered]
Hi Matt, 

Here in this example, we have ```nn1``` MLP neural network for message passing in ```NNConv```. 

``` Python
        nn1 = nn.Sequential(nn.Linear(2, 25), nn.ReLU(),
                            nn.Linear(25, 1 * 32))
        self.conv1 = NNConv(d.num_features, 32, nn1, aggr='mean') 
```
Is it possible to use other type of neural nets e.g. CNN, GNN, ... for ```nn1```?

Another question that I have is about the comparison between ```SplineConv``` and ```NNConv```. I think they have similar graph convolution methods except using other type of message passing. 
I think in many problems ```SplineConv``` works with much less time and memory required compared to a ```NNConv``` model that has similar accuracy. Then in which problems ```NNConv``` would be better than ```SplineConv```?

**Comment by rusty1s** [comment, marked as answer]
Yes, you can use any model as long as
* it expects input of shape `[*, edge_dim]` and maps to `[*, in_channels * out_channels]`
* it only operates on a single input tensor
The strength of `NNConv` is that it can easily handle many many edge features whereas `SplineConv` would fail.
